### PR TITLE
Add model validation and unit specs for models

### PR DESCRIPTION
**Pull request by Benmuiruri**
### Blog app - Add model validation and unit specs for models
Hello Reviewer :wave: , thank you for taking the time to review my PR. 

# What does this PR do?
:heavy_check_mark: Validate `name` and `posts_counter` in  `User` model
:heavy_check_mark: Validate `title`, `comments_counter`, `likes_counter` in `Post` model
:heavy_check_mark: Add tests for unit specs for `User` model: testing `name` validation, `posts_counter` validation and `last_three_posts` method
:heavy_check_mark: Add tests for unit specs for `Post` model: testing `title` validation, `comments_counter` validation, `comments_counter` validation and `last_five_comments` method

# Description of the completed tasks 
- Add validations for `User` model
- Add validations for `Post` model
- Add unit specs for all of `User` and `Post` models' methods and validations.
 
# How should it be tested
- checkout the branch
`git checkout validation-testing`
- run `rake` to run tests

### Thanks for your time and effort to review my code.

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (elisha2kyakpo1) [inline spec/models/comment_spec.rb]: - [ ] Kindly add tests for comments as well

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (elisha2kyakpo1) [inline spec/models/like_spec.rb]: - [ ] Let's add tests here as well.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Benmuiruri, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The PR title and description looks professional 👍
- You have a descriptive readme file 👍
- No linter errors 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  let(:user1) { User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Kenya.', posts_counter: 0) }
+  let(:post) do
```
Comment (SajjadAhmad14) [inline spec/models/comment_spec.rb:5]: Some specs are failing for Comment model. Kindly see the screenshot below for reference.
comment_error

File: spec/models/like_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  let(:user1) { User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Kenya.', posts_counter: 0) }
+  let(:post) do
+    Post.create(author: user1, title: 'Hello', text: 'This is my first post', likes_counter: 0, comments_counter: 0)
```
Comment (SajjadAhmad14) [inline spec/models/like_spec.rb:6]: Some specs are failing for Like model. Kindly see the screenshot below for reference
like_error
.

File: spec/models/post_spec.rb [hunk 5]
```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  let(:user1) { User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Kenya.', posts_counter: 0) }
```
Comment (SajjadAhmad14) [inline spec/models/post_spec.rb:4]: Some specs are failing for Post model. Kindly see the screenshot below for reference.
post_error

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ben, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Benmuiruri** [comment]
[SajjadAhmad14](https://github.com/SajjadAhmad14)  I have cloned the project and I have confirmed that the tests are running and passing.

**Review by DammyShittu — APPROVED** [review]
Hi @Benmuiruri,

## APPROVED 🟢

Great implementation of all the requirements in this milestone. 👏

✔️ All tests are passing

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**